catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +171,67 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        // test warnings
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String warning : ex.getWarnings()) {
+                assertThat(warning.equals(service.sanitize(warning)), is(true));
+            }
+        }
+        // test errors
+        try {
+            service.validate(getMetacard("dog_3leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String error : ex.getErrors()) {
+                assertThat(error.equals(service.sanitize(error)), is(true));
+            }
+        }
+        // sanitize is called twice for each validate call
+        verify(service, times(4)).sanitize(anyString());
+    }
+
+    @Test
+    public void testSanitization() throws SchematronInitializationException {
+        SchematronValidationService service = getService(false,
```

Comment:
I would personally prefer there to be at least a few more tests here. For example the whitespace related assertions, trim, nothing changed, etc. I generally like less assertions per test case. Perhaps groups certain assertions if they are all likely to fail in a single test case. I see your point @harrison-tarr you basically just re-use the service in each test, which I would agree with.